File: website/pumpConfig.php
<?php
class PumpConfig {

    private $config_file = "/etc/loggerd/user.var";

    public $rate;
    public $start;
    public $stop;
    public $on;
    public $rest;
    
    function __construct() {
     
        $fh = fopen($this->config_file, "r") or die("can't open file for reading");
        
        while (($line = fgets($fh, 4096)) !== false) {
                $argv = explode("=", $line);
                
                if(count($argv) > 1)
                {
                    if($argv[0] == "rate")
                       $this->rate = $argv[1];
                   
                    if($argv[0] == "start")
                       $this->start = $argv[1];
                   
                    if($argv[0] == "stop")
                       $this->stop = $argv[1]; 
                   
                    if($argv[0] == "on")
                       $this->on = $argv[1];   
                   
                    if($argv[0] == "rest")
                       $this->rest = $argv[1];
                }
        }
        
        fclose($fh);
    }
    
    function store($rate, $start, $stop, $on, $rest)
    {
        $fh = fopen($this->config_file, "w") or die("can't open file for writing");
        fwrite($fh, "rate=" . $rate . "\n");
        fwrite($fh, "start=" . $start . "\n");
        fwrite($fh, "stop=" . $stop . "\n");
        fwrite($fh, "on=" . $on . "\n");
        fwrite($fh, "rest=" . $rest . "\n");
        
        $this->rate = $rate;
        $this->start = $start;
        $this->stop = $stop;
        $this->on = $on;
        $this->rest = $rest;
        
        fclose($fh);
    }
    
}
?>

File: loggerd/src/pump_config.cpp
#include <string.h>
#include <stdlib.h>

#include "pump_config.h"

PumpConfig::PumpConfig()
{
    interval = -1;
    startHour = -1;
    stopHour = -1;
    runTime = -1;
    restTime = -1;

    mFD = fopen("/etc/loggerd/user.var", "r");
    if(!mFD)
    {
        printf("User variables could NOT be opened\n");
        return;
    }

    char line[64];
    while(fgets(line, 64, mFD))
        handleLine(line);
}

void PumpConfig::handleLine(char * line)
{
    char * p = strchr(line, '=');
    if(!p)
        return;

    char * arg = p + 1;
    *p = 0;
    //printf("%s -> %s\n", line, arg);

    if(!strncmp(line, "rate", 4))
    {
        interval = atoi(arg) * 60;
    }
    if(!strncmp(line, "start", 4))
    {
        startHour = atoi(arg);
    }
    if(!strncmp(line, "stop", 4))
    {
        stopHour = atoi(arg);
    }
    if(!strncmp(line, "on", 2))
    {
        runTime = atoi(arg) * 60;
    }
    if(!strncmp(line, "rest", 4))
    {
        restTime = atoi(arg) * 60;
    }
}

PumpConfig::~PumpConfig()
{
    if(mFD)
        fclose(mFD);
}

